_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,300 @@
+kubectl exec $(kubectl get pod -l app=sleep -n legacy -o jsonpath={.items..metadata.name}) -c sleep -n legacy -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+
+```
+200
+```
+
+
+## Setup end-user authentication
+
+You will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). Please follow the instruction [here](https://github.com/istio/istio/tree/master/security/tools/jwt) to create one. You can also use your own JWT/JWKS endpoint for the demo. Once you have that, let's export to some enviroment variables.
+
+
+```bash
+export JWKS=https://www.googleapis.com/service_accounts/v1/jwk/<YOUR-SVC-ACCOUNT>
+export TOKEN=<YOUR-TOKEN>
+```
+
+Also, for convenience, let's expose `httpbin.foo` via ingress (for more details, see [ingress task]({{home}}/docs/tasks/traffic-management/ingress.html)).
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: httpbin-ingress
+  namespace: foo
+  annotations:
+    kubernetes.io/ingress.class: istio
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /headers
+        backend:
+          serviceName: httpbin
+          servicePort: 8000
+EOF          
+```
+
+Get ingress IP
+```bash
+export INGRESS_HOST=$(kubectl get ing -n foo -o=jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}')
+```
+
+And run test query
+```bash
+curl $INGRESS_HOST/headers -s -o /dev/null -w "%{http_code}\n"
+```
+
+```
+{
+  "headers": {
+    "Accept": "*/*",
+    "Content-Length": "0",
+    "Host": "35.230.123.105",
+    "User-Agent": "curl/7.58.0",
+    "X-B3-Sampled": "1",
+    "X-B3-Spanid": "d729acfa46c072ba",
+    "X-B3-Traceid": "d729acfa46c072ba",
+    "X-Envoy-Internal": "true",
+    "X-Ot-Span-Context": "d729acfa46c072ba;d729acfa46c072ba;0000000000000000",
+    "X-Request-Id": "8f232322-7c73-9470-8481-5ab64233adf9"
+  }
+}
+```
+
+Now, let's add a policy that requires end-user JWT for `httpbin.foo`. The next command assume policy with name "httpbin" already exists (which should be if you follow previous sections). You can run `kubectl get policies.authentication.istio.io -n foo` to confirm, and use `istio create` (instead of `istio replace`) if resource is not found. Also note in this policy, peer authentication (mTLS) is also set, though it can be removed without affecting origin authencation settings.
+
+```bash
+cat <<EOF | istioctl replace -n foo -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "httpbin"
+spec:
+  targets:
+  - name: httpbin
+  peers:
+  - mtls:
+  origins:
+  - jwt:
+      issuer: "YOUR_SERVICE_ACCOUNT_EMAIL"
+      jwksUri: $JWKS
+  principalBinding: USE_ORIGIN
+EOF  
+```
+
+The same curl command before will return with 401 error code, as a result of sever is expecting JWT but none provide:
+```bash
+curl $INGRESS_HOST/headers -s -o /dev/null -w "%{http_code}\n"
+```
+```
+401
+```
+
+Attaching the valid token generate above will return success:
```

Comment:
the valid token generate -> the token generated
will return -> returns

Reply:
done